### PR TITLE
Analytics - add "blog_id" where appropriate

### LINKED ISSUE #4405: Capture blog/site ID as a parameter in WPAnalytics calls where appropriate
Find all calls to event tracking in WPAnalytics around the app and pass in blog/site ID if it's a .com or a Jetpack site when it makes sense.

cc: @daniloercoli

### PR DESCRIPTION
**Pull request by daniloercoli**
Add the blog_id property to Analytics if it's a .com or a Jetpack site, and when it makes sense for the tracked event.

Fix #4405

cc @astralbodies

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/WPLegacyEditPostViewController.m [hunk 1]
```diff
@@ -393,7 +394,7 @@ - (void)cancelEditing
     }
 
     if (![self.post hasUnsavedChanges]) {
-        [WPAnalytics track:WPAnalyticsStatEditorClosed];
+        [WPAnalytics track:WPAnalyticsStatEditorClosed withProperties:@{ WPAppAnalyticsKeyBlogID:[self.post blog].dotComID} ];
```
Comment (astralbodies) [inline WordPress/Classes/ViewRelated/Post/WPLegacyEditPostViewController.m]: It's cool to just chain with the dot operator - `self.post.blog.dotComID`

File: WordPress/Classes/Services/ReaderSiteService.m [hunk 2]
```diff
@@ -86,7 +87,7 @@ - (void)followSiteWithID:(NSUInteger)siteID success:(void(^)())success failure:(
             if (success) {
                 success();
             }
-            [WPAnalytics track:WPAnalyticsStatReaderSiteFollowed];
+            [WPAnalytics track:WPAnalyticsStatReaderSiteFollowed withProperties:@{ WPAppAnalyticsKeyBlogID:[NSNumber numberWithUnsignedInteger:siteID] }];
```
Comment (astralbodies) [inline WordPress/Classes/Services/ReaderSiteService.m]: I'd replace `[NSNumber numberWithUnsignedInteger:siteID]` with the newer syntactic sugar `@(siteID)`

**Comment by astralbodies** [comment]
Overall this looks good, @daniloercoli! I'd suggest you go through and only use the dot operator syntax when accessing properties instead of the classic object-message pattern (`post.blog.blogID` rather than `[post blog].blogID` etc).

File: WordPress/Classes/ViewRelated/Post/WPLegacyEditPostViewController.m (hunk 1 above)
Comment (daniloercoli) [inline WordPress/Classes/ViewRelated/Post/WPLegacyEditPostViewController.m]: Didn't know who I used a mixed pattern here.

**Comment by daniloercoli** [comment]
I think i've update all the references to blogID by using the dot operator syntax. Ready for review again.

**Comment by astralbodies** [comment]
Looks good @daniloercoli. Appears you'll have to merge in develop again. Ping me when you want the shipit squirrel!

**Comment by daniloercoli** [comment]
@astralbodies I've merged in develop. Ready to be merged once Travis finished checking the push.

**Comment by astralbodies** [comment]
:shipit: :smile: 

Looks like you'll need another develop merge but go ahead and just merge the PR when you're done.

#trust

**Pull request merged**